<?xml version="1.0" encoding="utf-8"?>
<robot name="dofbot">
    <link name="base_link">
        <visual>
            <origin
                    xyz="0 0 0"
                    rpy="0 0 0"/>
            <geometry>
                <mesh
                        filename="package://smart_headlamp_pkg/meshes/base_link.STL"/>
            </geometry>
            <material
                    name="">
                <color
                        rgba="0.7 0.7 0.7 1"/>
            </material>
        </visual>
        <collision>
            <origin
                    xyz="0 0 0"
                    rpy="0 0 0"/>
            <geometry>
                <mesh
                        filename="package://smart_headlamp_pkg/meshes/base_link.STL"/>
            </geometry>
        </collision>
    </link>
    <link name="link1">
        <visual>
            <origin
                    xyz="0 0 0"
                    rpy="0 0 0"/>
            <geometry>
                <mesh
                        filename="package://smart_headlamp_pkg/meshes/link1.STL"/>
            </geometry>
            <material
                    name="">
                <color
                        rgba="0 0.7 0 1"/>
            </material>
        </visual>
        <collision>
            <origin
                    xyz="0 0 0"
                    rpy="0 0 0"/>
            <geometry>
                <mesh
                        filename="package://smart_headlamp_pkg/meshes/link1.STL"/>
            </geometry>
        </collision>
    </link>
    <joint name="joint1" type="revolute">
        <origin
                xyz="0 0 0.064"
                rpy="0 0 0"/>
        <parent
                link="base_link"/>
        <child
                link="link1"/>
        <axis
                xyz="0 0 1"/>
        <limit
                lower="-1.5708"
                upper="1.5708"
                effort="100"
                velocity="1"/>
    </joint>
    <link name="link2">
        <visual>
            <origin
                    xyz="0 0 0"
                    rpy="0 0 0"/>
            <geometry>
                <mesh
                        filename="package://smart_headlamp_pkg/meshes/link2.STL"/>
            </geometry>
            <material
                    name="">
                <color
                        rgba="0.7 0.7 0.7 1"/>
            </material>
        </visual>
        <collision>
            <origin
                    xyz="0 0 0"
                    rpy="0 0 0"/>
            <geometry>
                <mesh
                        filename="package://smart_headlamp_pkg/meshes/link2.STL"/>
            </geometry>
        </collision>
    </link>
    <joint name="joint2" type="revolute">
        <origin
                xyz="0 0 0.0435"
                rpy="0 1.5708 0"/>
        <parent
                link="link1"/>
        <child
                link="link2"/>
        <axis
                xyz="0 0 1"/>
        <limit
                lower="-1.5708"
                upper="1.5708"
                effort="100"
                velocity="1"/>
    </joint>
    <!-- 직선(원기둥) 정의 -->
    <link name="marker_link">
        <visual>
            <geometry>
                <cylinder radius="0.01" length="7.0"/>
            </geometry>
            <origin xyz="-3.5 0 0" rpy="0 1.5708 0"/>
            <material name="direction">
                <color rgba="0.6 0.4 0.8 0.7"/>
            </material>
        </visual>
    </link>
    <!-- 직선의 끝점(joint) 정의 -->
    <joint name="marker_joint" type="fixed">
        <origin xyz="0 0 0" rpy="0 0 0"/>
        <parent link="link2"/>
        <child link="marker_link"/>
    </joint>
    
</robot>
